torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -95,11 +155,11 @@ def write_single_view(prefix: str, tensor: Tensor, base_index: int):
                 and len(base.size()) == len(tensor.size())
                 and base.size() == tensor.size()
             ):
-                kwargs[f"{prefix}_alias"] = True
+                use_alias()
+            elif (slice_info := try_use_slice(base, tensor)) is not None:
+                use_slice(*slice_info)
```

Comment:
nit: could be methods on the ViewInfos if we do the refactor I mentioned above, but not a big deal